walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -233,4 +240,44 @@
             MoreExecutors.directExecutor()
         );
     }
+
+    /**
+     * Gets all the applicant's interested jobs given the params.
+     *
+     * @param applicantId The applicant's userId.
+     * @param pageSize The the number of jobs to be shown on the page.
+     * @param pageIndex The page number on which we are at.
+     * @return Future of the JobPage object.
+     */
+    public Future<JobPage> fetchInterestedJobPage(String applicantId, int pageSize, int pageIndex) throws IOException {
+        CollectionReference applicantAccountsCollection = FireStoreUtils.getFireStore().collection(APPLICANT_ACCOUNTS_COLLECTION);
+
+        DocumentReference docRef = applicantAccountsCollection.document(applicantId);
+
+        return ApiFutures.transform(
+            docRef.get(),
+            documentSnapshot -> {
+                ImmutableList.Builder<Job> jobListBuilder = ImmutableList.builder();
+                List<String> interestedList = (List<String>) documentSnapshot.get("interestedJobs");
+
+                for (String jobId: interestedList) {
+                    try {
+                        Optional<Job> job = fetchJob(jobId).get(5, TimeUnit.SECONDS);
```

Comment:
you are doing what is called as a N+1 select. Meaning you do a select, then get a list, and for each item in the list you do another select. Are you able to batch select all the jobs instead by passing a list of Ids? (similar to a JOIN in SQL)?

https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping#:~:text=The%20N%2B1%20query%20problem,the%20larger%20the%20performance%20impact.